File: Producto/frontend/src/remote/DTO/ItemEVNDTO.js
/**
 * DTO para representar un Item de Evaluación de Negocio.
 * Basado en ItemEVN.java (3FN: campos descriptivos explícitos, no EAV)
 */
export class ItemEVNDTO {
    constructor(data = {}) {
        this.idItemEVN         = data.id || data.idItemEVN || null;
        this.nroItem           = data.nroItem || data.numero || null;
        this.articuloId        = data.articuloId || data.productoId || null;
        this.proveedorId       = data.proveedorId || null;
        this.proveedorNombre   = data.proveedorNombre || '';
        this.descripcion       = data.descripcion || '';
        this.modelo            = data.modelo || '';
        this.tela              = data.tela || '';
        this.composicion       = data.composicion || '';
        this.genero            = data.genero || '';
        this.codigoInterno     = data.codigoInterno || '';
        this.codigoProveedor   = data.codigoProveedor || '';
        this.cantidad          = data.cantidad || 0;
        this.precioUnitario    = data.precioUnitario || 0;
        this.costoUnitario     = data.costoUnitario || 0;
        this.costoProducto     = data.costoProducto || data.costoUnitario || 0;
        this.costoLogo         = data.costoLogo || 0;
        this.costoOrdenTrabajo = data.costoOrdenTrabajo || 0;
        this.costoTotalUnitario = data.costoTotalUnitario || 0;
        this.tipoItem          = data.tipoItem || '';
        // Array de specs verdaderamente dinámicas [{clave, valor}]
        this.technicalSpecs    = Array.isArray(data.technicalSpecs) ? data.technicalSpecs : [];
        this.margenItem        = data.margenItem || 0;
        this.totalItem         = data.totalItem || data.total || 0;
        this.costeoId          = data.costeoId || null;
        this.solicitudCostosId = data.solicitudCostosId || null;
    }
}

File: Producto/backend/src/main/java/backend/com/comercial/application/dto/ItemEVNDTO.java
package backend.com.comercial.application.dto;

import backend.com.comercial.domain.model.ItemEspecificacion;
import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Data
public class ItemEVNDTO {
    // IDs de relaciones (opcionales)
    private Integer articuloId;
    private Long proveedorId;

    // Datos descriptivos del ítem
    private Integer nroItem;
    private String descripcion;
    private String modelo;
    private String tela;
    private String composicion;
    private String genero;
    private String codigoInterno;
    private String codigoProveedor;
    private String proveedorNombre;

    // Cantidades y precios
    private Integer cantidad;
    private BigDecimal precioUnitario;
    private BigDecimal costoUnitario;

    // Costos individuales del ítem
    private BigDecimal costoProducto;
    private BigDecimal costoLogo;
    private BigDecimal costoOrdenTrabajo;

    // Clasificación
    private String tipoItem;

    // Especificaciones verdaderamente dinámicas (no conocidas de antemano)
    private List<ItemEspecificacion> technicalSpecs;

    // Vínculo con costeo/SCOS — solo si tipoItem = "OP"
    private Long costeoId;
    private Long solicitudCostosId;
}
